### PR TITLE
A stab at using vectors again.

### PR DESCRIPTION
**Pull request by hassila**
Please have a look at this PR, numbers looks good but needs a second eye for verifying correctness.
```
decode_direct4_class
=================================
2527 ms encode
0 ms decode
162 ms use
162 ms decode+use
=================================

decode_direct4_struct
=================================
2545 ms encode
0 ms decode
152 ms use
152 ms decode+use
=================================

decode_direct4_struct_with_vector
=================================
2525 ms encode
0 ms decode
136 ms use
136 ms decode+use
=================================
```

### CODE REVIEW

File: FBTest/FBReaderCommon.swift [hunk 1]
```diff
@@ -186,82 +186,96 @@ public extension FBReader {
...
-            nextIndex += 1
-            return value
+    public func index(after i: Int) -> Int {
+        return i+1
+    }
+    public subscript(i : Int) -> T {
+            return T(reader: reader, myOffset: reader.getVectorOffsetElement(vectorOffset: myOffset, index: i))!
+    }
+    /*
+    public subscript(i : Int) -> T? {
```
Comment (mzaks) [inline FBTest/FBReaderCommon.swift:271]: I would rather go with this subscript.
I generally made FlatBuffersSwift much more fault tolerant specifically on read.
Rationale is if some one sends you corrupt buffer intentionally, your application should not crash.
Specifically important for BackEnd applications

Comment (hassila) [reply FBTest/FBReaderCommon.swift:271]: The one returning an optional? Or the original one?

Comment (mzaks) [reply FBTest/FBReaderCommon.swift:271]: Returning optional is better the user can force unwrap it if they sure the buffer is not corrupted

Comment (hassila) [reply FBTest/FBReaderCommon.swift:271]: That's fine, it was my first implementation - it performs the same essentially.

Comment (mzaks) [reply FBTest/FBReaderCommon.swift:271]: 👍

**Comment by hassila** [comment]
Ok, f7a733c adopts that pattern. Interestingly enough the latest revision using vectors actually gives a consistent performance improvement of ~10%.

**Pull request merged**